parsec/scheduling.c

```diff
@@ -487,6 +487,9 @@ int __parsec_task_progress( parsec_execution_stream_t* es,
         if(task->status <= PARSEC_TASK_STATUS_HOOK) {
             rc = __parsec_execute( es, task );
         }
+        else if(task->status == PARSEC_TASK_STATUS_COMPLETE) { // To deal with task completion of the GPU tasks
```

Comment:
This code is unnecessary because the task returned from the GPU will be in the state `PARSEC_TASK_STATUS_COMPLETE`, which means the test `task->status <= PARSEC_TASK_STATUS_HOOK` will fail and the execution will do in the switch with rc set to `PARSEC_HOOK_RETURN_DONE` (because of the initialization at the beginning of the function), which translates into executing `__parsec_complete_execution()`.